lint/indexOnlyLoop_checker.go

```diff
@@ -0,0 +1,81 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+//! Detects when range loop is iterating over container which has pointer-type elements (that is, any *T or reference-like type as map or chan) using only key variable (no value) and container[key] occurs more than once inside loop body
+//
+// Suggests to use for key, v := range container form.
+//
+// @Before:
+// func closeFiles(files []*os.File) {
+//     for i := range files {
+//         if files[i] != nil {
+//            files[i].Close()
+//         }
+//     }
+// }
+//
+// @After:
+// func closeFilesSuggester(files []*os.File) {
+//     for _, f := range files {
+//         if f != nil {
+//             f.Close()
+//         }
+//     }
+// }
+
+func init() {
+	addChecker(&indexOnlyLoopChecker{}, attrExperimental)
+}
+
+type indexOnlyLoopChecker struct {
+	checkerBase
+}
+
+func (c *indexOnlyLoopChecker) VisitStmt(stmt ast.Stmt) {
+	if s, ok := stmt.(*ast.RangeStmt); ok && s.Key != nil {
+		sX := s.X.(*ast.Ident).Obj
+		var tp ast.Expr
+		switch sXField := sX.Decl.(type) {
```

Comment:
`s/sXField/decl/`? (optional.)

Reply:
seems that I don't get what you mean here...

Reply:
rename `sXField` to `decl` 😉

Reply:
already renamed to `sXFieldDecl`... I got it, it was simple sed...